src/sql/workbench/api/browser/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -362,6 +362,17 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 		return Promise.resolve(this.doOpenEditor(resource, options));
 	}
 
+	$trySetTrusted(uri: UriComponents, isTrusted: boolean): Promise<boolean> {
```

Comment:
Is there a reason why we can't leverage the notebookservice here instead of adding another API? Could we leverage the serializeNotebookStateChange() method (or refactor that method to also include another "updateTrust" method)?

Reply:
Do you mean extend the `NotebookService` to the extension via an RPC call? API is updated to invoke it from the extension. `NotebookService` is hosted in the main code base/workbench. Maybe I am missing something here